### ISSUE TITLE
Aln.convert.pl: XMFA -> FastA incomplete output

### ISSUE THREAD

**Issue by yseif** [lmrodriguezr/enveomics#34]
I'm using Aln.convert.pl to convert an XMFA file to a fasta format. In the process, a few sequences are removed and I end up with a smaller number of sequences than I started with. Is that normal?

**Comment by lmrodriguezr** [comment]
Hello @yseif 
`Aln.convert.pl` is simply a wrapper for Bio::Perl, so I'm inclined to think it's probably an issue with the parent function rather than this script. However, if you post an example of an input file causing the issue I can take a look and try to resolve it.

**Comment by AlisaGU** [comment]
I wrote a short test file. However, it ran a long time and no output. Do you know the reason?

here is the code:
perl Aln.convert.pl XMFA fasta test.xmfa test.fasta
[Uploading test.xmfa.zip…]()

and this is the test file:

#FormatVersion Mauve1
#Sequence1File      a.fa
#Sequence1Entry     1
#Sequence1Format    FastA
#Sequence2File      b.fa
#Sequence2Entry     2
#Sequence2Format    FastA
#Sequence3File      c.fa
#Sequence3Entry     3
#Sequence3Format    FastA
#BackboneFile       three.xmfa.bbcols
> 1:0-0 + a.fa
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
> 2:5417-5968 + b.fa
TTTAAACATCCCTCGGCCCGTCGCCCTTTTATAATAGCAGTACGTGAGAGGAGCGCCCTAAGCTTTGGGAAATTCAAGC-
--------------------------------------------------------------------------------
CTGGAACGTACTTGCTGGTTTCGCTACTATTTCAAACAAGTTAGAGGCCGTTACCTCGGGCGAACGTATAAACCATTCTG
> 3:9476-10076 - c.fa
TTTAAACACCTTTTTGGATG--GCCCAGTTCGTTCAGTTGTG-GGGAGGAGATCGCCCCAAACGTATGGTGAGTCGGGCG
TTTCCTATAGCTATAGGACCAATCCACTTACCATACGCCCGGCGTCGCCCAGTCCGGTTCGGTACCCTCCATGACCCACG
---------------------------------------------------------AAATGAGGGCCCAGGGTATGCTT
=
> 2:5969-6015 + b.fa
-----------------------
GGGCGAACGTATAAACCATTCTG
> 3:9429-9476 - c.fa
TTCGGTACCCTCCATGACCCACG
AAATGAGGGCCCAGGGTATGCTT
=

**Comment by AlisaGU** [comment]
I wrote a short test file. However, it ran a long time and no output. Do you know the reason?

here is the code:
perl Aln.convert.pl XMFA fasta test.xmfa test.fasta

**Comment by AlisaGU** [comment]
@lmrodriguezr I can't post the test .xmfa for unknown reasons. If you noticed this message, please tell me you email. I can deliver it to you

**Comment by lmrodriguezr** [comment]
Hello @AlisaGU 
The reason is that the script is waiting for input. Please check the correct syntax in the help message:
```
Aln.convert.pl in-format out-format < in_file > output_file
```

The `<` and the `>` are to redirect input from `in_file` and output to `output_file`, respectively. In your case, the command should be:

```
Aln.convert.pl XMFA fasta < test.xmfa > test.fasta
```

**Comment by AlisaGU** [comment]
I thought mistakenly “<>” just stands for there is a input file.

Thank you very much






| |
Shanshan Gu
|
|
邮箱：18845145385@163.com
|

Signature is customized by Netease Mail Master

On 07/07/2020 21:08, Luis M Rodriguez-R wrote:

Closed #34.
